client/apps/edit/components/section_container/index.coffee

```diff
@@ -37,23 +40,70 @@ module.exports = React.createClass
     e.stopPropagation()
     @props.section.destroy()
 
+  onDragStart: (e) ->
+    dragStart = e.clientY - ($(e.currentTarget).position().top - window.scrollY)
+    @props.onDragStart e, dragStart
+
+  onDragOver: (e) ->
+    mousePosition = e.clientY - @props.dragStart
+    $dragOver = $(e.currentTarget).find('.edit-section-container')
+    dragOverID = $dragOver.data('id')
+    dragOverTop = $dragOver.position().top + 20 - window.scrollY
+    dragOverCenter = dragOverTop + ($dragOver.height() / 2)
+    if mousePosition > dragOverCenter and dragOverID is !@props.sections.length or
+     dragOverID is @props.dragging + 1
+      @setState dropPosition: 'bottom'
+    else
+      @setState dropPosition: 'top'
+    @props.onSetDragOver dragOverID unless dragOverID is @props.dragOver
+
+  printDropPlaceholder: ->
+    if @props.dragOver is @props.index
+      unless @props.index is @props.dragging
+        div {
+          className: 'edit-section-drag-placeholder'
+          style: height: @props.draggingHeight
+        }
+
+  getContainerProps: ->
+    props = {}
+    unless @props.isHero
+      props = {
+        draggable: !@props.editing
+        onDragStart: @onDragStart
+        onDragEnd: @props.onDragEnd
+        onDragOver: @onDragOver
+        style: {'opacity': .65} if @props.dragging is @props.index
+      }
+    return props
+
   render: ->
-    div {},
+    div @getContainerProps(),
+      if @state.dropPosition is 'top' and !@props.isHero
+        @printDropPlaceholder()
```

Comment:
Nitpick—`print` is a little strange of a term as it usually implies something like `console.log`—maybe just `@dropPlaceholder()` or leaving the logic inside `render`?